packages/update-monitor/src/inversion/index.ts

```diff
@@ -43,17 +38,45 @@ export function invertAndPrint(
       }
       addresses.set(address, details)
     }
-    details.roles.push({ name: role, atName, atAddress })
+    if (role) {
+      details.roles.push(role)
+    }
   }
 
   for (const contract of project.contracts) {
-    for (const [key, value] of Object.entries(contract.values ?? {})) {
+    const upgradeabilityValues = contract.upgradeability as unknown as Record<
+      string,
+      ContractValue
+    >
+
+    add(contract.address)
+    const values: Record<string, ContractValue | undefined> = {
+      ...contract.values,
+      ...upgradeabilityValues,
+
+      // We don't want to show the implementation as nodes in the diagram
+      implementation: undefined,
+      callImplementation: undefined,
+      adminImplementation: undefined,
+      userImplementation: undefined,
+      implementations: undefined,
+    }
```

Comment:
We can make this a bit more robust by making sure we don't override values.
```suggestion
    const upgradeabilityValues = {
      ...contract.upgradeability as unknown as Record<string, ContractValue>

      // We don't want to show the implementation as nodes in the diagram
      implementation: undefined,
      callImplementation: undefined,
      adminImplementation: undefined,
      userImplementation: undefined,
      implementations: undefined,
    }

    add(contract.address)
    const values: Record<string, ContractValue | undefined> = {
      ...upgradeabilityValues,
      ...contract.values,
    }
```